tests/integration/test_postgresql.py

```diff
@@ -117,6 +118,8 @@ async def test_deploy_and_relate_postgresql(ops_test: OpsTest):
     await ops_test.model.applications[DATA_INTEGRATOR].remove_relation(
         f"{DATA_INTEGRATOR}:postgresql", f"{POSTGRESQL[ops_test.cloud_name]}:database"
     )
+    # Ensuring full cleanup of relation traces, avodiing faluire on re-creating it soon
+    sleep(3)
```

Comment:
The test above is destroying the relation that's re-created in the upcoming test.

This can lead to flaky behavior as Juju may not have had enough time to fully clean up traces of the previously existing relation. 
(If needed I can dig out the occasion when this happened on the pipeline.)

A small pause leaves better chances for the system to recover and to have stable pipelines.
(Note that the current :green_circle: is a 1st attempt. Didn't happen much on previous runs.)